pkg/storaged/mdraid/mdraid.jsx

```diff
@@ -91,22 +91,27 @@ function mdraid_stop(mdraid) {
 }
 
 function mdraid_delete(mdraid, block, card) {
-    function delete_() {
-        if (mdraid.Delete)
-            return mdraid.Delete({ 'tear-down': { t: 'b', v: true } }).then(reload_systemd);
-
-        // If we don't have a Delete method, we simulate
-        // it by stopping the array and wiping all
-        // members.
-
-        function wipe_members() {
-            return Promise.all(client.mdraids_members[mdraid.path].map(member => member.Format('empty', { })));
+    async function delete_() {
+        const members = client.mdraids_members[mdraid.path];
+
+        await mdraid.Delete({ 'tear-down': { t: 'b', v: true } });
+
+        // Now wipe the members again.  This is necessary when
+        // deleting a MDRAID with metadata 1.0, which stores its
+        // superblock at the end of each member, and the superblock of
+        // the actual content of the mdraid is stored at the very
+        // start of each member (for certain raid levels like a
+        // mirror).  The Delete call above will only remove the mdraid
+        // metadata and leave the content superblock in place. Thus,
+        // after deleting, the members might magically turn into the
+        // previous content.  Wiping the members again will remove any
+        // such metadata that might cause them to be misidentified.
```

Comment:
Did we make a udisks issue, as the commit says it is a udisks bug? If so we should link it here.